camus-etl-kafka/src/test/java/com/linkedin/camus/etl/kafka/CamusJobTestWithMock.java

```diff
@@ -170,17 +181,21 @@ private void setupJob1() {
     TopicMetadataResponse metadataResponse = EasyMock.createMock(TopicMetadataResponse.class);
     _mocks.add(metadataResponse);
     EasyMock.expect(metadataResponse.topicsMetadata()).andReturn(topicMetadatas).anyTimes();
+    return metadataResponse;
+  }
 
-    // For OffsetResponse
+  private OffsetResponse mockOffsetResponse(List<MyMessage> myMessages) {
     OffsetResponse offsetResponse = EasyMock.createMock(OffsetResponse.class);
     _mocks.add(offsetResponse);
-    List<MyMessage> myMessages = messagesWritten.get(TOPIC_1);
     // The first call is getLatestOffset, we set the value to 1
     EasyMock.expect(offsetResponse.offsets(EasyMock.anyString(), EasyMock.anyInt())).andReturn(new long[]{myMessages.size()}).times(1);
     // The second call is getEarliestOffset, we set the value to 0
     EasyMock.expect(offsetResponse.offsets(EasyMock.anyString(), EasyMock.anyInt())).andReturn(new long[]{0}).times(1);
+    EasyMock.expect(offsetResponse.hasError()).andReturn(false).anyTimes();
```

Comment:
Why not passing the expected return value as an argument to mockOffsetResponse(), this way you can control whether the mock object should return true or false.